File: examples/node-worker/enqueue.js
import { FeatherClient } from "@arbtrage/feather";

const client = new FeatherClient(process.env.FEATHER_ADDRESS ?? "localhost:50051");

async function main() {
  const count = Number(process.env.COUNT ?? 10);
  for (let i = 0; i < count; i++) {
    const { jobId } = await client.enqueue({
      name: "echo",
      payload: JSON.stringify({ n: i + 1 }),
    });
    console.log("enqueued", jobId);
  }
  client.close();
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: examples/python-worker/enqueue.py
#!/usr/bin/env python3
"""Enqueue sample jobs."""

import json
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "..", "..", "packages", "sdk-python"))

from getfeather import FeatherClient


def main() -> None:
    client = FeatherClient()
    count = int(os.environ.get("COUNT", "10"))
    for i in range(count):
        job_id = client.enqueue(
            "echo",
            payload=json.dumps({"n": i + 1}).encode(),
        )
        print("enqueued", job_id)
    client.close()


if __name__ == "__main__":
    main()
